src/js/components/status/LiveStatusIndicator.jsx

```diff
@@ -80,17 +80,18 @@ export class LiveStatusIndicator extends Component {
 
             // update the progress each second
             this.clearIntervalId = setInterval(() => {
-                this._updateProgress();
+                this._updateProgress(this.props);
             }, 1000);
 
-            this._updateProgress();
+            this._updateProgress(props);
         }
     }
 
-    _updateProgress() {
+    _updateProgress(props: Props) {
         const now = moment();
         const elapsed = now.diff(this.startTime);
-        this.percentage = Math.floor(elapsed / this.props.estimatedDuration * 100);
+        const estimatedDuration = props && props.estimatedDuration || this.props.estimatedDuration;
```

Comment:
Reworked this a bit in 8f4554a to only calculate the percentage if we receive valid numbers, otherwise just put the spinner to indeterminate.